### PR TITLE
Fix embedded conf assignment when loading trace or model

### LINKED ISSUE #41: Cannot load FGS trace after loading musical chairs model
The following sequence of commands results in unexpected behaviour:

```
(aldb) load models/musical_chairs.als
Reading model from models/musical_chairs.als...done.
(aldb) setconf traces/fgs_conf.yml
Setting parsing options from traces/fgs_conf.yml...done.
(aldb) trace traces/fgs_counterexample.xml
Reading trace from traces/fgs_counterexample.xml...error. Sig State not found.
error. Invalid trace.
```

### PR DESCRIPTION
**Pull request by liangdrew**
Closes #41.

### CODE REVIEW

**Comment by pxlcoder** [comment]
I don’t think this is the correct approach to solving the bug. Instead, I believe the embedded conf should be cleared in `initializeWithTrace`, similar to how it is cleared in `initializeWithModel` if an embedded conf is not present.

The issue with the current approach is that the user will be unable to step through the currently loaded model after using `setconf`. Since this command is only responsible for setting a persistent configuration, it should not have any other side effects.

**Comment by liangdrew** [comment]
Thanks @pxlcoder – that's a great suggestion. That seems better to me as well.

**Review by aman-dureja — APPROVED** [review]

File: src/simulation/SimulationManager.java [hunk 1]
```diff
@@ -185,6 +185,9 @@ public boolean initializeWithModel(File model) {
     }
 
     public boolean initializeWithTrace(File trace) {
+        // Ensure any embedded ParsingConf from a previously loaded model is removed.
+        embeddedParsingConf = null;
```
Comment (pxlcoder) [inline src/simulation/SimulationManager.java:83]: I don't think this code should run if the trace fails to be loaded.

File: src/simulation/SimulationManager.java [hunk 2]
```diff
@@ -528,6 +398,155 @@ public String getWorkingDirPath() {
...
+        this.traceMode = false;
+        this.activeSolutions.clear();
+        this.activeSolutions.push(sol);
+
+        return true;
+    }
+
+    private boolean initializeWithTrace(File trace) {
+        // Ensure any embedded ParsingConf from a previously loaded model is removed.
+        embeddedParsingConf = null;
```
Comment (pxlcoder) [inline src/simulation/SimulationManager.java]: Can you move this to the `initialize` method?

File: src/simulation/SimulationManager.java [hunk 3]
```diff
@@ -528,6 +398,155 @@ public String getWorkingDirPath() {
...
+            System.out.println("error. Failed to read file.");
+            return false;
+        }
+
+        String configString = ParsingConf.getConfStringFromFileString(modelString).trim();
+        if (configString.isEmpty()) {
+            // If the new model has no embedded ParsingConf, make sure any existing embedded
+            // ParsingConf for the previous model is removed.
+            embeddedParsingConf = null;
+        } else {
```
Comment (pxlcoder) [inline src/simulation/SimulationManager.java]: This can just become:

```
if (!configString.isEmpty()) {
    try {}
}
```

**Review by pxlcoder — APPROVED** [review]

**Pull request merged**